src/backend/gitalytics_api/routes/get_commits_per_day.py

```diff
@@ -3,31 +3,45 @@
 r"""
 
 """
+import typing as t
 import fastapi
 import pydantic
+import sqlalchemy as sql
+import sqlalchemy.orm
 from database import models as dbm
 from database.db import createLocalSession
-import sqlalchemy as sql
+from gitalytics_api.cookies import active_workspace_id
 
 
 router = fastapi.APIRouter()
 
 
-class ResponseModel(pydantic.BaseModel):
-    commits_per_day: dict[str, int]
+@fastapi.Depends
+def get_database_connection():
+    db = createLocalSession()
+    try:
+        yield db
+    finally:
+        db.close()
 
 
-@router.get("/commits_per_day/", response_model=ResponseModel)
-async def get_commits_per_day(workspace_id: int, year: int):
+@router.get("/commits-per-day", response_model=t.Dict[str, int])
+async def get_commits_per_day(
+        connection: sql.orm.Session = get_database_connection,
+        workspace_id: int = active_workspace_id,
+        year: int = fastapi.Query()
+):
     r"""
     get commits per day in the workspace from a specific year
     """
-    with createLocalSession() as connection:
-        commits_per_day = connection.query(sql.func.date(dbm.Commit.committed_at), sql.func.count()) \
-            .group_by(sql.func.date(dbm.Commit.committed_at)) \
-            .join(dbm.Repository.commits) \
-            .filter(dbm.Repository.workspace_id == workspace_id) \
-            .filter(sql.func.extract('year', dbm.Commit.committed_at) == year) \
-            .all()
-
-    return {"commits_per_day": commits_per_day}
+    repos = connection.query(dbm.Repository.id).filter(dbm.Repository.workspace_id == workspace_id)
```

Comment:
wie in dem anderen Kommentar begründet, die Repos müssen über die Session geholt werden.